### ISSUE TITLE
How to configure fans which only has increase/decrease buttons?

### ISSUE THREAD

**Issue by felipecrs** [smartHomeHub/SmartIR#980]
This is not a bug, but rather a question. The thing is, I see many fans in the default codes but all of them seem to support remotes that has specific buttons for each speed. In my case, I have a remote which only has the "increase" and "decrease" buttons. I wonder how I should configure SmartIR to handle this case.

**Comment by dankarization** [comment]
All these remotes also has increase and decrease buttons, not numbers. But the thing is- codes they send really don't send "increase speed" command, but rather specific set of rules. In fact, any of the remotes button do this.

You can check it this way: set your AC to some mode, then block your remote (just block IR light by hand) and press "increase" a few times. Then open the light and press "increase" 1 more time. If your AC will go to the same more which your remote displays- it's supported by smartir and does right what I explained before

**Comment by felipecrs** [comment]
My remote is actually RF and not IR. Maybe that explains.

**Comment by dankarization** [comment]
Same principal applies usually

**Comment by iKaew** [comment]
Hi, I got the same situation with this issue. 

The remote (IR) has a single button for on and increase speed (3 levels). If the current speed is at level 3 the speed will go back to level 1 again. 

I try to make a PR to support this device, but not sure in the json file, what should be set in `speed` and default command ? Since there is no specific code for level 1,2,3

**Comment by Dakyne** [comment]
Same issue here with Honeywell fan. The remote has a On/Off button, swing button and speed button. When pressing speed it increases and loop back to slow when the fastest speed has been reached.

**Comment by felipecrs** [comment]
> All these remotes also has increase and decrease buttons, not numbers. But the thing is- codes they send really don't send "increase speed" command, but rather specific set of rules. In fact, any of the remotes button do this.
> 
> You can check it this way: set your AC to some mode, then block your remote (just block IR light by hand) and press "increase" a few times. Then open the light and press "increase" 1 more time. If your AC will go to the same more which your remote displays- it's supported by smartir and does right what I explained before

Yeah, so it turns out that it's not the case for me. I think both my controller and my fan aren't smart enough for this. I had to resort to some tricky solutions like a Python script for controlling the fan speed, by iterating through the number of times it has to press the button (https://github.com/iml885203/HA-FanSpeedControl).

But I think it would be great if this feature could be part of SmartIR so more people could benefit of it.

**Comment by sergiobaiao** [comment]
> My remote is actually RF and not IR. Maybe that explains.
> 
> > If your AC will go to the same more which your remote displays- it's supported by smartir and does right what I explained before
> 
> My controller hasn't a display.
> 
> ![image](https://user-images.githubusercontent.com/29582865/208313899-7eb42ac6-5a89-41f2-b58a-d5a20b985c98.png)
> 
> Thanks a lot for the explanation though.
> 
> I guess, as the name implies, I can only use SmartIR with IR controlled devices, lol.

I control the same device with a Broadlink RM Pro 3 and it works with SmartIR

**Comment by felipecrs** [comment]
@vassilis-panos is there some news about this feature or did you just close it because it's old?
